wurst/systems/meatsystem/MeatSystem.wurst

```diff
@@ -0,0 +1,379 @@
+    if meatInventoryInstances.has(u)
+        meatInventoryInstances.get(u).tryPickupMeatArea()
+    else
+        let instance = new MeatInventory(u)
+        meatInventoryInstances.put(u, instance)
+        instance.tryPickupMeatArea()
+
+function onTargetOrder()
+    let order = GetIssuedOrderId()
+    let u = GetOrderedUnit()
+    if order != Orders.smart
+        return
+
+    let target = GetOrderTargetItem()
+
+    //Smart pickup system -> simulate picking up meat like an item if full inventory, like in Bannars StackNSplit system
+    if (target != null
+        and target.isPickupable()
+        and target.getTypeId() == ITEM_RAW_MEAT
+        and u.isInventoryFull()
+        and (meatInventoryInstances.has(u) or isUnitMeatCarrier(u)))
+
+        if meatInventoryInstances.has(u)
+            let instance = meatInventoryInstances.get(u)
+            let meatCount = instance.carriedMeat.size()
+            if meatCount < MAX_MEAT_COUNT
+                instance.onSmartPickupMeat(target)
+        else
+            let instance = new MeatInventory(u)
+            meatInventoryInstances.put(u, instance)
+            instance.onSmartPickupMeat(target)
+
+function isUnitMeatCarrier(unit u) returns bool
+    return UnitEntity.findForUnit(u) instanceof Troll or u.getTypeId() == UNIT_ENSNARE_TRAP
+
+init
+    //Handle picking up meat normally to inventory
+    registerPlayerUnitEvent(EVENT_PLAYER_UNIT_PICKUP_ITEM) ->
+        let maybeMeat = GetManipulatedItem()
+        let u = GetTriggerUnit()
+        if maybeMeat.getTypeId() == ITEM_RAW_MEAT
+            u.removeItem(maybeMeat)
+            if isUnitMeatCarrier(u)
+                unitTryPickupMeat(u, maybeMeat)
+
+    //Setup smooth pickup event (ability to pick up raw meat with full inventory)
+    registerPlayerUnitEvent(EVENT_PLAYER_UNIT_ISSUED_TARGET_ORDER) ->
+        onTargetOrder()
+
+    //Death event
+    registerPlayerUnitEvent(EVENT_PLAYER_UNIT_DEATH) ->
+        let dying = GetTriggerUnit()
+        if meatInventoryInstances.has(dying)
+            meatInventoryInstances.get(dying).onOwnerDeath()
+
+    //Setup pickup and drop abilities
+    for i = 0 to 7
+        registerSpellEffectEvent(ABILITY_IDS_DROPMEAT[i]) ->
+            let caster = GetSpellAbilityUnit()
+            if meatInventoryInstances.has(caster)
+                meatInventoryInstances.get(caster).emptyToGround()
+
+    registerSpellEffectEvent(ABILITY_ID_PICKUPMEAT) ->
+        let caster = GetSpellAbilityUnit()
+        if meatInventoryInstances.has(caster) or isUnitMeatCarrier(caster)
+            unitTryPickupMeatArea(caster)
+
+    registerSpellEffectEvent(ABILITY_ID_PICKUPMEAT_TROLL) ->
+        let caster = GetSpellAbilityUnit()
+        if meatInventoryInstances.has(caster) or isUnitMeatCarrier(caster)
+            unitTryPickupMeatArea(caster)
+
+    //Special fix for beastmaster to get his abilities back after transforming
+    for s in ABILITY_IDS_SHAPESHIFTER
```

Comment:
Can you avoid doing this manually, counting the new system? If the ability is permanent it should transfer between transformation automatically.

Reply:
WIth new BM system we dont need this.